tests/arch/x86/early_serial/src/main.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+
+extern int arch_printk_char_out(int c);
+extern void __printk_hook_install(int (*fn)(int));
+
+char *expected = "22 113 10000 32768 40000 22\n"
+		 "p 112 -10000 -32768 -40000 -22\n"
+		 "0x1 0x01 0x0001 0x00000001 0x0000000000000001\n"
+		 "0x1 0x 1 0x   1 0x       1\n"
+		 "42 42 0042 00000042\n"
+		 "-42 -42 -042 -0000042\n"
+		 "42 42   42       42\n"
+		 "42 42 0042 00000042\n"
+		 "255     42    abcdef        42\n"
+		 "68719476735 -1 18446744073709551615 ffffffffffffffff\n"
+		 "0xcafebabe 0xbeef 0x2a\n";
```

Comment:
Is the expected string has any meaning? What we want to check here? It seems like we only judge the "successfully" here.

Reply:
yes. this strings has no meaning. I'll remove it.

Reply:
Excuse me, I made you misunderstand. I just want to know if this string should also be checked. You don't have to remove them, as long as it's meaningful and be checked, thanks.

Reply:
this string needn't be check, it just write here to send out to console. Because  "Successful! " is also send to the console and be checked, the long string can be remove.

Reply:
You could keep them if they are meaningful, I think.